### PR TITLE
Update Existing PR

### PR DESCRIPTION
**Pull request by AbhitejJohn**
Added in an ability to update an existing PR as a fix for #396. If one chooses to overwrite it would now be a flag one can specify with the update schematics.

Also fixed the possibility of resetting to a stale base branch with a Fetch.

This might need a change to certain Release definitions - if someone can point me to these I'll go fix them up as well.

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 1]
```diff
@@ -281,13 +281,14 @@ private static async Task<string> DownloadArtifactsAsync(BuildHttpClient buildCl
...
                 // Be judicious and clean up old artifacts so we do not eat up memory on the scheduler machine.
-                Directory.Delete(tempDirectory, recursive: true);
-                
-                // Sometimes a creation of a directory races with deletion since at least in .net 4.6 deletion is not a blocking call.
-                // Hence explictly waiting for the directory to be deleted before moving on.
-                Stopwatch w = Stopwatch.StartNew();
+                //Directory.Delete(tempDirectory, recursive: true);
 
-                while (Directory.Exists(tempDirectory) && w.ElapsedMilliseconds < 20 * 1000) Thread.Sleep(100);
+                //// Sometimes a creation of a directory races with deletion since at least in .net 4.6 deletion is not a blocking call.
```
Comment (brettfo) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: Comment cleanup.

File: src/RoslynInsertionTool/scripts/UpdateExistingPr.ps1 [hunk 2]
```diff
@@ -16,7 +16,8 @@ param([string] $enlistmentPath,
       [string] $updateAssemblyVersions,
       [string] $updateCoreXTLibraries,
       [string] $visualStudioBranchName,
-      [string] $writePullRequest)
```
Comment (brettfo) [inline src/RoslynInsertionTool/scripts/UpdateExistingPr.ps1]: Just a note to remember to update the Azure DevOps release definition when this is in to take into account the script name change.  I can help with this if you need.

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 1 above)
Comment (AbhitejJohn) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: Sorry, this was not meant to be. I'll undo this change.

File: src/RoslynInsertionTool/scripts/UpdateExistingPr.ps1 (hunk 2 above)
Comment (AbhitejJohn) [reply src/RoslynInsertionTool/scripts/UpdateExistingPr.ps1]: yes, that would be helpful.

Comment (AbhitejJohn) [reply src/RoslynInsertionTool/scripts/UpdateExistingPr.ps1]: Synced up offline. Will make a change.

**Pull request merged**